java-advanced-training/src/main/java/com/epam/java_advanced_training/handler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.java_advanced_training.handler;
+
+import com.epam.java_advanced_training.exception.ProductNotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+  @ExceptionHandler(ProductNotFoundException.class)
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  public String handleProductNotFoundException(ProductNotFoundException ex) {
+    return ex.getMessage();
```

Comment:
There is a good practice to build and return a specific error type with information about the error case instead of returning just a string.